backend/src/main/java/gov/cdc/usds/simplereport/service/PersonService.java

```diff
@@ -499,8 +541,13 @@ public Person updatePatient(
   }
 
   @AuthorizationConfiguration.RequirePermissionArchiveTargetPatient
-  public Person setIsDeleted(UUID patientId, boolean deleted) {
-    Organization patientOrg = _os.getCurrentOrganization();
+  public Person setIsDeleted(UUID patientId, boolean deleted, String orgExternalId) {
+    Organization patientOrg;
+    if (orgExternalId != null && StringUtils.isNotEmpty(orgExternalId)) {
```

Comment:
[StringUtils.IsNotEmpty](https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/StringUtils.html#:~:text=to%20isNotBlank(CharSequence)-,isNotEmpty,-public%20static%C2%A0boolean) actually does a null check as well so you can get rid of your null check!